subworkflows/local/polishing/pilon/run_pilon/main.nf

```diff
@@ -0,0 +1,20 @@
+include { PILON } from '../../../../../modules/nf-core/pilon/main'
+workflow RUN_PILON {
+  take:
+  assembly_in
+  aln_to_assembly_bam_bai
+
+  main:
+  assembly_in
+    .join(aln_to_assembly_bam_bai)
+    .set { pilon_in }
+
+  PILON(
+    pilon_in.map { it -> [it[0], it[1]] },
```

Comment:
Ideally, so others can also follow, be more explicit with your variable names.
Avoid using `it`, and when closures have identifiable elements the split the input i.e. `{ meta, bam -> ... }`